swim/join_sender.go

```diff
@@ -388,7 +388,11 @@ func (j *joinSender) JoinGroup(nodesJoined []string) ([]string, []string) {
 					failed = true
 					break
 				}
+
 				j.node.memberlist.Update(res.Membership)
+				for _, change := range res.Membership {
```

Comment:
This is a clever way of solving this problem. I was thinking that it could/should be solved by making the initial membership-set more explicit in the API, but for the sake of time, I'm OK with this. I will ask you to place a comment about the ClearChange loop saying why we're doing this.